kimchi/js/bindings.js

```diff
@@ -2199,11 +2226,43 @@ var caml_pasta_fp_commitments_to_rust = function (x) {
   var w_comm = convertArray(x[1]);
   var z_comm = caml_vesta_poly_comm_to_rust(x[2]);
   var t_comm = caml_vesta_poly_comm_to_rust(x[3]);
-  return new plonk_wasm.WasmFpProverCommitments(w_comm, z_comm, t_comm);
+  var lookup;
+  if (x[4] === 0) {
+    lookup = undefined;
+  } else {
+    lookup = caml_fp_lookup_commitments_to_rust(x[4][1]);
+  }
+  return new plonk_wasm.WasmFpProverCommitments(w_comm, z_comm, t_comm, lookup);
 };
 
-// Provides: caml_pasta_fp_commitments_of_rust
+// Provides: caml_fp_lookup_commitments_of_rust
 // Requires: caml_vesta_poly_comm_of_rust, js_class_vector_of_rust_vector, plonk_wasm
+var caml_fp_lookup_commitments_of_rust = function (wasm_lc) {
+  var convertArray = function (v) {
+    var a = js_class_vector_of_rust_vector(v, plonk_wasm.WasmFpPolyComm);
+    var res = [0];
+    for (var i = 0; i < a.length; ++i) {
+      // TODO Check this. Could be off by 1
+      res.push(caml_vesta_poly_comm_of_rust(a[i]));
+    }
```

Comment:
Maybe rewrite as
``` 
a.forEach((elt) => { res.push(caml_vesta_poly_comm_of_rust(elt)); })
```

Reply:
Will be available with a new version of jsoo. Could be changed with https://github.com/MinaProtocol/mina/pull/13777 as jsoo seems now to be more es6 friendly, see https://github.com/ocsigen/js_of_ocaml/pull/1391